webapp-client/src/component/Navbar.js

```diff
@@ -8,31 +9,44 @@ export default class Navbar extends React.Component {
       loggedIn: false,
       authInstance: null,
     };
+    allowErrorMessage(this);
     this.formatUser = this.formatUser.bind(this);
     this.gapiSetState = this.gapiSetState.bind(this);
     this.setLoggedIn = this.setLoggedIn.bind(this);
     this.signOut = this.signOut.bind(this);
+    this.onAuth2Init = this.onAuth2Init.bind(this);
   }
 
   componentDidMount() {
     window.gapi.load('auth2', this.gapiSetState);
   }
 
   gapiSetState() {
-    window.gapi.auth2.init({
-      client_id: process.env.REACT_APP_CLIENT_ID,
-    });
+    window.gapi.auth2
+      .init({
+        client_id: process.env.REACT_APP_CLIENT_ID,
+      })
+      .then(
```

Comment:
can you do await instead?

Reply:
I don't know the intricacies of promises, but I made the change and it seems like it works. Any pointers are appreciated!